extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java

```diff
@@ -107,30 +131,33 @@ protected synchronized void stopScan() {
 
     @Override
     public void onLightAdded(@Nullable HueBridge bridge, FullLight light) {
-        onLightAddedInternal(light);
+        onObjectAddedInternal(light);
     }
 
-    private void onLightAddedInternal(FullLight light) {
-        ThingUID thingUID = getThingUID(light);
-        ThingTypeUID thingTypeUID = getThingTypeUID(light);
+    private void onObjectAddedInternal(FullHueObject object) {
+        ThingUID thingUID = getThingUID(object);
+        ThingTypeUID thingTypeUID = getThingTypeUID(object);
 
-        String modelId = light.getModelID().replaceAll(HueLightHandler.NORMALIZE_ID_REGEX, "_");
+        String modelId = object.getModelID().replaceAll(HueLightHandler.NORMALIZE_ID_REGEX, "_");
 
         if (thingUID != null && thingTypeUID != null) {
             ThingUID bridgeUID = hueBridgeHandler.getThing().getUID();
             Map<String, Object> properties = new HashMap<>(1);
-            properties.put(LIGHT_ID, light.getId());
+            properties.put(LIGHT_ID, object.getId());
             properties.put(MODEL_ID, modelId);
-            properties.put(LIGHT_UNIQUE_ID, light.getUniqueID());
+            String uniqueID = object.getUniqueID();
+            if (uniqueID != null) {
```

Comment:
There is a [virtual daylight sensor](https://developers.meethue.com/documentation/supported-sensors#daylightSensor) on the bridge which does not have an unique id, because it is implemented by the bridge itself. therefore the `@NonNull` annotation on `getUniqueID` is technically wrong.